lib/utils/logger.js

```diff
@@ -214,6 +216,47 @@ const log = method => (...messages) => {
 	logMessages.forEach(msg => logLine(msg, consoleMethod, fontColor, leftRail));
 };
 
+/**
+ * Create logging stream for console logs from device.
+ * When console WS event is received from device, data is normalized and logged to custom Console stdout/stderr,
+ * data in those streams is transformed (prefixes added to each line) and piped to current process stdout.
+ */
+function createDeviceLogger() {
+	let logTime = 0;
+
+	// Create transform stream to prepend our custom prefix to each output new line.
+	const transformStream = new Transform({
+		transform(chunk, encoding, callback) {
+			const prefix = `${stColors.gray(timestamp.formatDate(logTime) + devicePrefix())}${stColors.bold('AL:')} `;
+
+			chunk = prefix + String(chunk).replace(/\n$/, '').replace(/\n/gm, '\n' + prefix) + '\n';
+
+			callback(null, chunk);
+		},
+	});
+
+	// Enable console colors support in node 10.16.2 and higher
+	const options = semver.gte(process.version, '10.16.2') ? [{
```

Comment:
According to docs we can use any of these options, did not find any limitations by node versions.

Reply:
We have node v8 as a requirement for js api usage in package json and for code climate we require node v9.6. Those earlier versions do not support colorMode. Code climate failed when I tried to use object as argument, thats how I noticed it.